app/code/Magento/InventoryCatalog/Plugin/InventoryApi/SetDataToLegacyCatalogInventoryAtSourceItemsSavePlugin.php

```diff
@@ -99,7 +103,7 @@ public function afterExecute(SourceItemsSaveInterface $subject, $result, array $
             $this->setDataToLegacyStockStatus->execute(
                 $sourceItem->getSku(),
                 (float)$sourceItem->getQuantity(),
-                (int)$this->isProductSalable->execute($sourceItem->getSku(), $this->defaultStockProvider->getId())
+                (int)$this->isProductSalable->execute($sourceItem->getSku(), $defaultSalesChannel)
```

Comment:
this looks incorrect to me
```
    public function afterExecute(SourceItemsSaveInterface $subject, $result, array $sourceItems)
    {
        $defaultSalesChannel = $this->salesChannelByWebsiteIdProvider->execute(0);

       //... 

            $this->setDataToLegacyStockStatus->execute(
                $sourceItem->getSku(),
                (float)$sourceItem->getQuantity(),
                (int)$this->isProductSalable->execute($sourceItem->getSku(), $defaultSalesChannel)
            );
        }
    }
```

Before that logic looked as 
`(int)$this->isProductSalable->execute($sourceItem->getSku(), $this->defaultStockProvider->getId())`

and our business idea was to check whether Product is salable in Default Stock.

Currently `$defaultSalesChannel = $this->salesChannelByWebsiteIdProvider->execute(0);`
There is no guarantee that Main website would be assigned to Default stock.

Need to think whether we need to have two  Interfaces with `SalesChannel` and `Stock ` as parameters